lib/gsup/server.py

```diff
@@ -216,3 +221,22 @@ async def __handle_gsup(self, peer: IPAPeer, payload_length: int):
         if request is None:
             raise ValueError(f"Error parsing GSUP message from peer {peer}")
         await self.gsup_handler.dispatch(peer, request)
+
+    async def _listen_for_subscriber_updates(self):
+        """
+        Listens for subscriber update events on a Redis queue and processes them.
+        """
+        await self.logger.logAsync(service='GSUP', level='INFO', message="Listening for subscriber updates")
+        while True:
+            try:
+                _, message_data = await asyncio.to_thread(self.redis_messaging.awaitMessage, 'subscriber_update')
+                update_data = json.loads(message_data)
+                update_event = SubscriberInfo(**update_data)
+
+                await self.logger.logAsync(service='GSUP', level='INFO',
+                                           message=f"Received subscriber update for IMSI {update_event.imsi} with new MSISDN {update_event.msisdn}")
+                await self.gsup_handler.dispatch_subscriber_update(update_event)
+
+            except Exception as e:
+                await self.logger.logAsync(service='GSUP', level='ERROR',
+                                           message=f"Error processing subscriber update: {e}")
```

Comment:
How about logging a full traceback here too on error?

```suggestion
                                           message=f"Error processing subscriber update: {traceback.format_exc()}")
```

Otherwise it is really hard to understand some errors, e.g. I just got this because I didn't have redis running while testing the PR:

```
[11/20/2025 10:12:46] [ERROR] Error processing subscriber update: not enough values to unpack (expected 2, got 0)
```

With the traceback:

```
[11/20/2025 10:14:28] [ERROR] Error processing subscriber update: Traceback (most recent call last):
  File "/home/user/.cache/osmo-ttcn3-testenv/host/install/venv/lib/python3.12/site-packages/pyhss/lib/gsup/server.py", line 232, in _listen_for_subscriber_updates
    _, message_data = await asyncio.to_thread(self.redis_messaging.awaitMessage, 'subscriber_update')
    ^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 2, got 0)
```